src/document.c

```diff
@@ -449,8 +445,12 @@ FIELD_PREPROCESSOR(fulltextPreprocessor) {
       aCtx->tokenizer->Start(aCtx->tokenizer, (char *)c, fl, options);
 
       Token tok = {0};
-      uint32_t newTokPos;
-      while (0 != (newTokPos = aCtx->tokenizer->Next(aCtx->tokenizer, &tok))) {
+      while (0 != aCtx->tokenizer->Next(aCtx->tokenizer, &tok)) {
+        if ((!strcmp(tok.tok, "")) && !FieldSpec_IndexesEmpty(fs)) {
+          // Skip empty values if the field should not index them
+          // Empty tokens are returned only if the original value was empty
+          continue;
+        }
```

Comment:
Can move out of the loop?
```suggestion
      bool isIndexEmpty = FieldSpec_IndexesEmpty(fs);
      while (0 != aCtx->tokenizer->Next(aCtx->tokenizer, &tok)) {
        if (!isIndexEmpty && (!strlen(tok.tok))) {
          // Skip empty values if the field should not index them
          // Empty tokens are returned only if the original value was empty
          continue;
        }
```

Reply:
Good idea in general.
But the `isIndexEmpyt` test must come after the test for the empty string, otherwise we will just not index any token in the case of non-empty indexing field :)